Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -86,10 +86,14 @@ private class ParameterSetNames
 
         /// <summary>
         /// <para type="description">
-        /// The ID of the dataset that you want to get a descriptor object for.
+        /// The ID of the dataset that you want to get a descriptor object for. This field also accepts 
+        /// DatasetData objects so they can be mapped to full Dataset objects.
         /// </para>
         /// </summary>
-        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.Get)]
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.Get)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetList.DatasetsData),
+            Property = nameof(DatasetList.DatasetsData.DatasetReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetReference), Property = nameof(DatasetReference.DatasetId))]
         public string Dataset { get; set; }
```

Comment:
Consider:

    gcloud config set core/project your-project
    $whatTables = Get-BqDataset -Project "my-project" | Get-BqTable

`$whatTables` will now contain Tables from "your-project" even though the command was piped in Datasets from "my-project". Users will find that very surprising. Instead, we need to have two parameter sets, one that takes a -Project [string] and a -DatasetId [string], and one that takes a -Dataset [DatasetReference], because DatasetReference has both project and datasetId